test/data/BinaryHeap.ts

```diff
@@ -0,0 +1,495 @@
+          await instance['add(bytes32)'](twoBytes32);
+
+          expect(await instance['indexOf(bytes32)'](zeroBytes32)).to.equal(1);
+          expect(await instance['indexOf(bytes32)'](oneBytes32)).to.equal(2);
+          expect(await instance['indexOf(bytes32)'](twoBytes32)).to.equal(0);
+        });
+
+        it('returns max uint256 if value does not exist', async () => {
+          expect(await instance['indexOf(bytes32)'](zeroBytes32)).to.equal(
+            ethers.constants.MaxUint256.toString(),
+          );
+        });
+      });
+
+      describe('#length()', () => {
+        it('returns length of binary heap', async () => {
+          expect(await instance['length()']()).to.equal(0);
+
+          await instance['add(bytes32)'](zeroBytes32);
+          expect(await instance['length()']()).to.equal(1);
+
+          await instance['add(bytes32)'](oneBytes32);
+          expect(await instance['length()']()).to.equal(2);
+
+          await instance['add(bytes32)'](twoBytes32);
+          expect(await instance['length()']()).to.equal(3);
+
+          await instance['remove(bytes32)'](twoBytes32);
+          expect(await instance['length()']()).to.equal(2);
+
+          await instance['remove(bytes32)'](oneBytes32);
+          expect(await instance['length()']()).to.equal(1);
+
+          await instance['remove(bytes32)'](zeroBytes32);
+          expect(await instance['length()']()).to.equal(0);
+        });
+      });
+
+      describe('#root()', () => {
+        it('returns value of the root node', async () => {
+          await instance['add(bytes32)'](zeroBytes32);
+          expect(await instance['root()']()).to.equal(zeroBytes32);
+
+          await instance['add(bytes32)'](oneBytes32);
+          expect(await instance['root()']()).to.equal(oneBytes32);
+
+          await instance['add(bytes32)'](twoBytes32);
+          expect(await instance['root()']()).to.equal(twoBytes32);
+
+          await instance['remove(bytes32)'](twoBytes32);
+          expect(await instance['root()']()).to.equal(oneBytes32);
+
+          await instance['remove(bytes32)'](oneBytes32);
+          expect(await instance['root()']()).to.equal(zeroBytes32);
+        });
+
+        describe('reverts if', function () {
+          it('index out of bounds', async () => {
+            await expect(instance['root()']()).to.be.revertedWith(
+              'EnumerableHeap: index out of bounds',
+            );
+          });
+        });
+      });
+
+      describe('#add(bytes32)', () => {
+        it('adds value to heap', async () => {
+          await instance['add(bytes32)'](zeroBytes32);
+          await instance['add(bytes32)'](oneBytes32);
+          await instance['add(bytes32)'](twoBytes32);
+
+          expect(await instance['heap()']()).to.deep.equal([
+            twoBytes32,
+            zeroBytes32,
+            oneBytes32,
+          ]);
+        });
+      });
+
+      describe('#remove(bytes32)', () => {
```

Comment:
Add test for root being what it should be given the `_maxHeapify` sorting.

Also, for the nodes to be ordered in accordance with said sorting.

Reply:
Perhaps the latter of these are superfluous given the `#root` tests check implicitly for these. Thoughts?

Reply:
> Perhaps the latter of these are superfluous given the #root tests check implicitly for these. Thoughts?

I believe this is true, sorting can only be checked if an element is removed from the tree therefore I can only check that the tree is sorted by calling something like `root` to make sure the head of the tree is the highest value. It may make sense to check that the following property holds true though `in which the value of a node is greater than or equal to the values of its children`.

Reply:
If that property is satisfied, then for sure the algorithm is working correctly and all above suggestions in this comment thread (root check eg) are needless.

Reply:
added in https://github.com/solidstate-network/solidstate-solidity/pull/134/commits/0d02343d3377301915ef9617f5780e79420e3032